src/VirtualClient/VirtualClient.Dependencies/NvidiaGPUDriverInstallation.cs

```diff
@@ -0,0 +1,149 @@
+using VirtualClient.Common.Extensions;
+using VirtualClient.Common.Telemetry;
+using VirtualClient.Contracts;
+
+namespace VirtualClient.Dependencies
+{
+    /// <summary>
+    /// Provides functionality for installing Nvidia GPU driver.
+    /// </summary>
+    public abstract class NvidiaGPUDriverInstallation : VirtualClientComponent
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NvidiaGPUDriverInstallation"/> class.
+        /// </summary>
+        /// <param name="dependencies"></param>
+        /// <param name="parameters"></param>
+        public NvidiaGPUDriverInstallation(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters = null)
+            : base(dependencies, parameters)
+        {
+            this.RetryPolicy = Policy.Handle<Exception>().WaitAndRetryAsync(5, (retries) => TimeSpan.FromSeconds(retries + 1));
+        }
+
+        /// <summary>
+        /// A policy that defines how the component will retry when
+        /// it experiences transient issues.
+        /// </summary>
+        public IAsyncPolicy RetryPolicy { get; set; }
+
+        /// <summary>
+        /// The version of GPU driver to be installed.
+        /// </summary>
+        public string DriverVersion
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(NvidiaGPUDriverInstallation.DriverVersion), string.Empty);
+            }
+
+            set
+            {
+                this.Parameters[nameof(NvidiaGPUDriverInstallation.DriverVersion)] = value;
+            }
+        }
+
+        /// <summary>
+        /// Determines whether Reboot is required or not after Driver installation.
+        /// </summary>
+        public bool RebootRequired
+        {
+            get
+            {
+                switch (this.Platform)
+                {
+                    case PlatformID.Unix:
+                    case PlatformID.Win32NT:
+                        return this.Parameters.GetValue<bool>(nameof(NvidiaGPUDriverInstallation.RebootRequired), false);
+
+                    default:
+                        return this.Parameters.GetValue<bool>(nameof(NvidiaGPUDriverInstallation.RebootRequired), true);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Checks if Nvidia driver is installed and logs its version.
+        /// </summary>
+        /// <param name="telemetryContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        protected async Task<bool> CheckIfDriverInstalledAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            string driverVersionCommand = this.Platform == PlatformID.Unix ? "-c \"nvidia-smi --query-gpu=driver_version --format=csv,noheader\"" : "nvidia-smi --query-gpu=driver_version --format=csv,noheader";
```

Comment:
I fixed a small bug here, where I forgot to add quotes around the nvidia-smi command after bash -c